### ISSUE TITLE
Navigation: Add keyboard navigation to menus and submenus

### ISSUE THREAD

**Issue by axelavargas** [grafana/grafana#38454]
**What happened:** 

The first level navigation submenus are not accessible via the keyboard, there are some [guides](https://www.w3.org/WAI/tutorials/menus/flyout/) from WCAG to address this problem.

But unfortunately, this guide is misaligned with our current design:

- Our menu items are also a link that redirects to a different page
- We don't have a button that serves as a visual indicator for the presence of a submenu.

Alternatives to solve the problem:

- Improve our UX and add a toggle button for submenus -- Navigation is through tabs, we will use flyout menus
- Follow a different approach to make the menu accessible, something like treating our navigation menu as a [Web Application Menu](https://www.w3.org/WAI/tutorials/menus/application-menus-code/) -- Here navigation is through arrows keys, and not using tab tab

**What you expected to happen:** 

- [ ] User is able to navigate the first level menu using keyboard (Tab)
- [ ] User visually is aware there are submenus when navigating first level navigation (When tabbing the submenu fly out)
- [ ] User is able to navigate inside submenu using arrow keys, Arrow Right-> Focus first item submenu, Arrows up and down -> Navigate submenu
- [ ] User is not forced to move through all submenus to switch to another first-level menu, it can use Esc, Tab, or the left arrow key
- [ ] User should be able to move away from the menu to main section (Skip links)
- [ ] Make sure this work doesn't interfere with Enterprise
    - [ ] Check that it compiles with Enterprise
    - [ ] Check with someone on Enterprise that we haven't broken anything 

**How to reproduce it (as minimally and precisely as possible):**
